src/backend/access/transam/xact.c

```diff
@@ -1739,13 +1739,13 @@ RecordTransactionAbort(bool isSubXact)
 	 */
 	TransactionIdAbortTree(xid, nchildren, children);
 
+	END_CRIT_SECTION();
```

Comment:
I ran into this error while running a test. It happened when a transaction was aborted and the `CSNSnapshotAbort` needed to load a page from the pageserver. This code path involved a call to `palloc`, which asserted that it is not in a critical section. 

Now that I think about it. This call should not load a page from the pageserver because we're writing to the local region. This seems to be a side effect of other bugs that messed with the regions and lsns. I'll revert this part and test everything again with the other fixes.

Reply:
Done

Reply:
Thanks for fixing this!